### PR TITLE
[BE][Endpoint] Add New Endpoint for Incrementing Game's viewCount

### PR DESCRIPTION
**Pull request by grgsferry**
## Description
**DOD:**
- to create new endpoint (`POST: /api/v1/games/view-count/:id/`)
- with no authentication and no HTTP BODY
- with response
  - ```json
    ‌{
        "success": true,
        "message": "Success Update View Count",
        "data": {
          "id": 1
          "viewCount": 600
        },
        "error": null
    }
    ```

**NOTE:** this endpoint will be hitted by FE whenever user/player open Game Details Page

- Trello Link: [here](https://trello.com/c/5pyTzlpw/18-beendpoint-add-new-endpoint-for-incrementing-games-viewcount)

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor/Breaking Change

### CODE REVIEW

**Review by hrz8 — APPROVED** [review]

File: src/domains/games/repository.js [hunk 1]
```diff
@@ -5,4 +5,14 @@ module.exports = {
     const allGames = await Game.findAll();
     return allGames;
   },
+  addViewCount: async (id) => {
+    const updatedGame = await Game.findOne({ where: id });
+    updatedGame.viewCount += 1;
```
Comment (irvanariyanto) [inline src/domains/games/repository.js:16]: aku coba masukin `id` yang ngga ada di database kena error mas, mungkin bisa ditambahin condition jika `updatedGame` nya null

File: src/domains/games/router.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ const gameController = require('./controller');
 
 const routes = {
   'GET: /': [authentication, authorization(['Player', 'Admin']), validation(gameValidator.list), gameController.list],
+  'POST: /view-count/:id': [gameController.addViewCount],
```
Comment (Nursalli) [inline src/domains/games/router.js]: Mungkin dsini bisa nambah validator mas untuk paramsnya

File: src/domains/games/repository.js [hunk 3]
```diff
@@ -5,4 +5,14 @@ module.exports = {
     const allGames = await Game.findAll();
     return allGames;
   },
+  addViewCount: async (id) => {
+    const updatedGame = await Game.findOne({ where: id });
```
Comment (Nursalli) [inline src/domains/games/repository.js:10]: Disini kasih kondisi if jg mas jika game id nya tidak ditemukan

File: src/domains/games/router.js (hunk 2 above)
Comment (hrz8) [reply src/domains/games/router.js]: thank you mas udah jeli, aku missed hehe @Nursalli 

```js
gameId: z.string().regex(/^\d+$/).transform(Number)
```

ref:https://github.com/colinhacks/zod/discussions/330

cc @grgsferry

Comment (grgsferry) [reply src/domains/games/router.js]: done added mas, thankyou!

File: src/domains/games/repository.js (hunk 3 above)
Comment (grgsferry) [reply src/domains/games/repository.js:10]: done added jg mas! thankyou

**Review by hrz8 — APPROVED** [review]

File: src/domains/games/repository.js [hunk 4]
```diff
@@ -5,4 +5,19 @@ module.exports = {
...
+
+    if (!updatedGame) {
+      throw new AppError('Game Not Found', 404);
```
Comment (irvanariyanto) [inline src/domains/games/repository.js:13]: `AppError` nya sepertinya lupa di import mas `const { AppError } = require('../../utils/error');`

**Review by irvanariyanto — CHANGES_REQUESTED** [review]

File: src/domains/games/repository.js (hunk 4 above)
Comment (grgsferry) [reply src/domains/games/repository.js:13]: done mas irvan! thankyou

File: src/domains/games/repository.js (hunk 1 above)
Comment (grgsferry) [reply src/domains/games/repository.js:16]: di commit "[Add viewCount validation and error if game not found](https://github.com/Nursalli/game-project-service/pull/8/commits/1b9f81e6c9ede580e35b2cb7fd10d955e3cb0a7e)" udah kutambah mass, thankyou

**Review by irvanariyanto — APPROVED** [review]

**Pull request merged**